### ISSUE TITLE
Do not fail img[alt][role=button]

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#2582]
**Expected**: The following should not fail axe-core:

```html
<img alt="hello world" role="button" />
```

**Actual**: This element fails the button-name rule.

**Comment by WilcoFiers** [comment]
Duplicate of #2577
